pkg/operator/staticpod/controller/prune/prune_controller.go

```diff
@@ -0,0 +1,304 @@
+package prune
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"github.com/openshift/library-go/pkg/operator/staticpod/controller/common"
+	"github.com/openshift/library-go/pkg/operator/staticpod/controller/prune/bindata"
+)
+
+// PruneController is a controller that watches static installer pod revision statuses and spawns
+// a pruner pod to delete old revision resources from disk
+type PruneController struct {
+	targetNamespace, podResourcePrefix          string
+	failedRevisionLimit, succeededRevisionLimit int
+
+	// command is the string to use for the pruning pod command
+	command []string
+	// queue only ever has one item, but it has nice error handling backoff/retry semantics
+	queue workqueue.RateLimitingInterface
+	// prunerPodImageFn returns the image name for the pruning pod
+	prunerPodImageFn func() string
+
+	operatorConfigClient common.OperatorClient
+
+	kubeClient    kubernetes.Interface
+	eventRecorder events.Recorder
+
+	dynamicClient      dynamic.Interface
+	operatorConfigGVR  schema.GroupVersionResource
+	operatorConfigName string
+}
+
+const (
+	pruneControllerWorkQueueKey = "key"
+	statusConfigMapName         = "revision-status-"
+	defaultResourceDir          = "/etc/kubernetes/static-pod-resources"
```

Comment:
making magic values into constants does not improve code quality.  do this if it is used in many places or possible to change. this is neither.  Just inline